laravel/app/AutoAssignment.php

```diff
@@ -176,19 +194,69 @@ private static function assignResidentsForPref($residents, $prefNum, $ticketsToA
                     }
                 }
                 // assign winning resident to schedule
-                self::addAssignment($wantedSchedule, $winnerResident, $date);
-                self::updateProbability($winnerResident, $ticketsToAdd);
+                $anestPref = Option::where('date', $date)
+                                ->where('schedule', $wantedSchedule)
+                                ->where('option', $prefNum)
+                                ->where('isValid', "1")
+                                ->where('resident', $winnerResident)
+                                ->value('anesthesiologist_id');
+                $anestsAssigned = self::handleAssignment($wantedSchedule, $winnerResident, $date, $ticketsToAdd, $anestPref, $anestsAssigned);
             }
         }
-        return $remainder;
+        return [$remainder,$anestsAssigned];;
+    }
+
+    // handles a resident's assignment checking if anest can be assigned and updating tickets 
+        // schedule = id for schedule 
+        // resident = id for resident
+        // date = date for schedule in YYYY-MM-DD format
+        // ticketsToAdd = initial number of tickets that will be added to resident's total
+        // anestPref = id for anesthesiologist
+        // anestsAssigned = array of assigned anests where key=anest_id and value=# of assignments
+    private static function handleAssignment($schedule, $resident, $date, $ticketsToAdd, $anestPref, $anestsAssigned) {
+        // if resident has an anesthesiologist pref - check if anest can be assigned
+        if(!is_null($anestPref)){
+            // get room
+            $room = ScheduleData::where('date', $date)
+                        ->where('id', $schedule)
+                        ->value('room');
```

Comment:
Doesn't an `id` uniquely identify a ScheduleData? Why can't this just be `ScheduleData::find($schedule)->room`?